internal/provider/networks_devices_claim_resource_test.go

```diff
@@ -0,0 +1,102 @@
+package provider
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// TestAccNetworksDevicesClaimResource function is used to test the CRUD operations of the Terraform resource you are developing.
+// It runs the test cases in order to create, read, update, and delete the resource and checks the state at each step.
+func TestAccNetworksDevicesClaimResource(t *testing.T) {
+
+	// The resource.Test function is used to run the test cases.
+	resource.Test(t, resource.TestCase{
+		// PreCheck function is used to do the necessary setup before running the test cases.
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+
+		// Steps is a slice of TestStep where each TestStep represents a test case.
+		Steps: []resource.TestStep{
+
+			// Create and Read an Organization.
+			{
+				Config: testAccNetworksDevicesClaimResourceConfigCreateOrganization,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("meraki_organization.test", "id", "example-id"),
+					resource.TestCheckResourceAttr("meraki_organization.test", "name", "test_meraki_networks_devices_claim"),
+				),
+			},
+
+			// Create and Read a Network.
+			{
+				Config: testAccNetworksDevicesClaimResourceConfigCreateNetwork,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("meraki_network.test", "name", "Main Office"),
+					resource.TestCheckResourceAttr("meraki_network.test", "timezone", "America/Los_Angeles"),
+					resource.TestCheckResourceAttr("meraki_network.test", "tags.#", "1"),
+					resource.TestCheckResourceAttr("meraki_network.test", "tags.0", "tag1"),
+					resource.TestCheckResourceAttr("meraki_network.test", "product_types.#", "3"),
+					resource.TestCheckResourceAttr("meraki_network.test", "product_types.0", "appliance"),
+					resource.TestCheckResourceAttr("meraki_network.test", "product_types.1", "switch"),
+					resource.TestCheckResourceAttr("meraki_network.test", "product_types.2", "wireless"),
+					resource.TestCheckResourceAttr("meraki_network.test", "notes", "Additional description of the network"),
+				),
+			},
+
+			// Claim and Read NetworksDevicesClaim
+			{
+				Config: testAccNetworksDevicesClaimResourceConfigCreate(os.Getenv("TF_ACC_MERAKI_MX_SERIAL")),
```

Comment:
ACC tests look great, Only thing is to add the ENV VARS for the MX Serial to the repos secrets.